share/src/main/java/org/alfresco/web/site/servlet/MTAuthenticationFilter.java

```diff
@@ -85,7 +81,35 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
             requestHolder.remove();
         }
     }
-    
+
+    private String getLanguageFromRequestHeader(HttpServletRequest req)
+    {
+        Locale locale = Locale.getDefault();
+        String acceptLang = req.getHeader(ACCEPT_LANGUAGE_HEADER);
+        if (acceptLang != null && acceptLang.length() > 0)
+        {
+            StringTokenizer tokenizer = new StringTokenizer(StringUtils.stripUnsafeHTMLTags(acceptLang), ",; ");
+            // get language and convert to java locale format
+            String language = tokenizer.nextToken().replace('-', '_');

```

Comment:
Make sure tokenizer.nextToken() doesn't trow exception, or if it throws catch it and set the locale to default.